service/storage_service.cc

```diff
@@ -187,12 +187,27 @@ std::optional<gms::inet_address> storage_service::get_replace_address() {
 }
 
 bool storage_service::is_replacing() {
-    sstring replace_address_first_boot = _db.local().get_config().replace_address_first_boot();
-    if (!replace_address_first_boot.empty() && _sys_ks.local().bootstrap_complete()) {
+    const auto& cfg = _db.local().get_config();
+    if (!cfg.replace_node_first_boot().empty()) {
+        if (_sys_ks.local().bootstrap_complete()) {
+            slogger.info("Replace node on first boot requested; this node is already bootstrapped");
+            return false;
+        }
+        return true;
+    }
+    if (!cfg.replace_address_first_boot().empty()) {
+      if (_sys_ks.local().bootstrap_complete()) {
         slogger.info("Replace address on first boot requested; this node is already bootstrapped");
         return false;
-    }
-    return bool(get_replace_address());
+      }
+      return true;
+    }
+    // Returning true if cfg.replace_address is provided
+    // will trigger an exception down the road if bootstrap_complete(),
+    // as it is an error to use this option post bootstrap.
```

Comment:
I understand this is a pre-existing problem, but maybe we should just throw an exception right here. (optional)

BTW. the documentation of "replace_address" is misleading when compared to "replace_address_first_boot", it suggests that we might be able to replace using a node that has already bootstrapped... Did this ever actually work? Did it actually work in Cassandra? It doesn't make any sense to me.

Reply:
I'm not sure how it worked.
It's discouraged and I'm not sure why we even bother to support it anymore.